core/src/main/java/org/infinispan/statetransfer/StateConsumerImpl.java

```diff
@@ -250,6 +256,18 @@ public boolean isStateTransferInProgressForKey(Object key) {
       return distributionInfo.isWriteOwner() && !distributionInfo.isReadOwner();
    }
 
+   @Override
+   public long inflightRequestCount() {
+      synchronized(transferMapsLock) {
+         return transfersBySegment.size();
```

Comment:
@wburns  should we include the transaction count here as well? Something like

```suggestion
         return transfersBySegment.size() + inflightTransactionSegmentCount();
```

Reply:
No, there is no clear distinction as to what this value is then. Having separate values is much cleaner in that you can see if there are transactions left that are being requested or if it is segment data. The way you have it with two different values is better.